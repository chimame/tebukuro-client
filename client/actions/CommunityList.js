import { createAction } from 'redux-actions'
import { getCommunities } from '../api/Community'

export const CREATE_COMMUNITIES = 'CREATE_COMMUNITIES'

export const createCommunities = createAction(CREATE_COMMUNITIES, getCommunities)
