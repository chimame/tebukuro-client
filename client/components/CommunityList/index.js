import React from 'react'
import { Link } from 'react-router'

const CommunityList = () => {
  return (
    <div>
      <div>community list sample</div>

      <Link to="/communities/new">new</Link>
    </div>
  )
}

export default CommunityList
