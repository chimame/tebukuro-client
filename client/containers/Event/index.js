import React, { Component }   from 'react'
import { connect }            from 'react-redux'
import { bindActionCreators } from 'redux'

export class Event extends Component {
  render() {
    return (
      <div>
        event page sample
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Event)
