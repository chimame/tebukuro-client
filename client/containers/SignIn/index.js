import React from 'react'
import { connect } from 'react-redux'
import OAuth from '../../components/OAuth'

const mapStateToProps = (state, ownProps) => {
  return {
    auth: state.auth
  }
}

export default connect(mapStateToProps, {})(OAuth)
