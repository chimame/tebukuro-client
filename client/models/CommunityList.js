import { Record, Set } from 'immutable'

const CommunityListRecord = Record({
  communities: Set.of()
})

export default class CommunityList extends CommunityListRecord {
  setCommunities(list) {
    return this.set('communities', Set.of(...list))
  }
}
