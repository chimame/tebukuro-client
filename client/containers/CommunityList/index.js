import React, { Component }   from 'react'
import { connect }            from 'react-redux'
import { bindActionCreators } from 'redux'
import { Link }               from 'react-router'
import * as Actions           from '../../actions/CommunityList'
import CommunityListTable     from '../../components/CommunityList'

export class CommunityList extends Component {
  componentDidMount() {
    this.props.actions.createCommunities()
  }

  render() {
    return (
      <div>
        <CommunityListTable CommunityList={this.props.communityList} />

        <Link to="/communities/new">new</Link>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    communityList: state.CommunityList
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators(Actions, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CommunityList)
