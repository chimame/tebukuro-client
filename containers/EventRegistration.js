// @flow
import React from 'react'
import { connect } from 'react-redux'
import withProvider from '../components/Provider'
import EventFormComponent from '../components/EventForm'
import { createEvent } from '../actions/event'
import { validateEventForm } from '../actions/eventForm'
import { getEventErrorsArray } from '../selectors/event'
import { getValidationErrorMessages, isValidationFailed } from '../selectors/eventForm'

type EventFormProps = {
  createEvent: Function,
  validateEventForm: Function,
  validationErrors: Object,
  validationFailed: boolean,
  errors: ?string[]
}

const EventRegistration = (props: EventFormProps) => (
  <div>
    <h3>This is the event form page!</h3>
    { props.errors && (
      <ul>
        { props.errors.map(error =>
          <li>{ error }</li>)}
      </ul>
    ) }
    <EventFormComponent
      onSubmit={props.createEvent}
      onValidation={props.validateEventForm}
      validationErrors={props.validationErrors}
      validationFailed={props.validationFailed}
    />
  </div>
)

const mapStateToProps = state => ({
  errors: getEventErrorsArray(state),
  validationErrors: getValidationErrorMessages(state),
  validationFailed: isValidationFailed(state),
})

const mapDispatchToProps = {
  createEvent,
  validateEventForm,
}

export default withProvider(connect(mapStateToProps, mapDispatchToProps)(EventRegistration))
