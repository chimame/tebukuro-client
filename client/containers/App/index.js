import React, { Component } from 'react'
import SignIn from '../SignIn'

export default class App extends Component {
  render() {
    return (
      <div>
        <SignIn />
        {this.props.children}
      </div>
    )
  }
}
