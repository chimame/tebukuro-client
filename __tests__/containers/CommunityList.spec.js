import React              from 'react'
import { mount, shallow } from 'enzyme'
import ToJson             from 'enzyme-to-json'
import { CommunityList }  from '../../client/containers/CommunityList'
import CommunityModel     from '../../client/models/Community'
import CommunityListModel from '../../client/models/CommunityList'

const communities = [new CommunityModel({id: 1, name: "name1"}), new CommunityModel({id: 2, name: "name2"})]
const communityList = new CommunityListModel({communities: communities})

const props = {
  communityList: communityList,  
  actions: { createCommunities: jest.fn() }
}

describe('CommunityList container', () => {

  it('should call createCommunities in componentDidMount once', () => {
    const wrapper = mount(<CommunityList {...props} />)
    expect(props.actions.createCommunities).toHaveBeenCalledTimes(1)
  })

  it('should render self and a subcomponent', () => {
    const wrapper = shallow(<CommunityList {...props} />)
    const tree = ToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
})
