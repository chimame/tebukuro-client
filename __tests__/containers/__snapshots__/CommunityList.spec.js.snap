// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityList container should render self and a subcomponent 1`] = `
<div>
  <CommunityList
    CommunityList={
      Object {
        "communities": Array [
          Object {
            "description": "",
            "errors": Array [],
            "id": 1,
            "name": "name1",
          },
          Object {
            "description": "",
            "errors": Array [],
            "id": 2,
            "name": "name2",
          },
        ],
      }
    }
  />
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/communities/new"
  >
    new
  </Link>
</div>
`;
